src/TLuaInterpreter.cpp

```diff
@@ -2554,57 +2554,104 @@ int TLuaInterpreter::setExitStub(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#connectExitStub
 int TLuaInterpreter::connectExitStub(lua_State* L)
 {
-    int toRoom;
-    int roomsGiven = 0;
-    int roomId = getVerifiedInt(L, __func__, 1, "fromID");
+    int toRoom = 0;
+    bool hasDirection = false;
+    bool hasToRoomId = false;
+    int fromRoom = getVerifiedInt(L, __func__, 1, "fromID");
 
-    int dirType = dirToNumber(L, 2);
-    if (!dirType) {
-        lua_pushfstring(L, "connectExitStub: bad argument #2 type (toID as number or direction as number or string expected, got %s!)");
-        return lua_error(L);
-    }
-    if (!lua_isnumber(L, 3) && !lua_isstring(L, 3)) {
-        roomsGiven = 0;
-    } else {
-        roomsGiven = 1;
-        toRoom = lua_tonumber(L, 2);
-        dirType = dirToNumber(L, 3);
-        if (!dirType) {
-            lua_pushstring(L, "connectExitStub: Invalid direction entered.");
-            return lua_error(L);
-        }
-    }
     Host& host = getHostFromLua(L);
-    if (!host.mpMap) {
-        return 0;
+    if (!host.mpMap || !host.mpMap->mpRoomDB) {
+        return warnArgumentValue(L, __func__, "no map present or loaded");
     }
-    TRoom* pR = host.mpMap->mpRoomDB->getRoom(roomId);
-    if (!pR) {
-        lua_pushstring(L, "connectExitStub: RoomId doesn't exist");
-        return lua_error(L);
+    if (lua_gettop(L) < 2) {
+        lua_pushfstring(L, "connectExitStub: missing argument #2 (toID as number or direction as number or string expected)");
```

Comment:
```suggestion
        lua_pushfstring(L, "connectExitStub: missing argument #2 (toID as number or direction number/string expected)");
```

@Kebap your thoughts on the wording?

Reply:
That's not how we do things around here anymore. We now have functions to phrase the error messages in order to unify the user experience and easy developer workflow. 
However in case of these overloaded functions, which can accept either number or string in a same position, the functions are not yet smart enough. So we need to still manually craft the error message to mimic the automatic formula.
You can test any function and leave an obligatory argument to see the error message reads like this:
`tempLineTrigger: bad argument #2 type (how many lines to match for as number expected, got 
no value!)`
So for your case we land round about here:
```suggestion
        lua_pushfstring(L, "connectExitStub: bad argument #2 type (toID as number, or direction as number/string expected, got no value!)");
```
I got no opinion on whether "or" or slash is better to inform both number and string are possible. I would not drop "as" to seperate argument name and type as that's a pretty established pattern.

Reply:
TBH I hadn't looked too closely into the situation where one gets a `no value` as a/an (*non-existent*) argument type! Like Kebap says, I did have to manually construct the error message because I couldn't get the required message out of the functions because the `getVerifiedXxxx(...)` ones are only constructed for a single type - and they try to coerce the provided argument to that type - whereas this one needed to do the exact opposite (i.e. only treat it as a number if it is one and a string likewise). In this case saying that the argument is missing is simpler to read though I agree that it doesn't follow how our messages have tended to go - perhaps we might want to look at our handling of missing *required* arguments more generally?

This one can be revisited I am sure - but I think we benefit more from the immediate improvement in functionality of this PR as it is then we might lose from stalling as we argue over this detail.